Factory/PsrHttpFactory.php

```diff
@@ -67,12 +69,28 @@ public function createRequest(Request $symfonyRequest)
 
         $body = $this->streamFactory->createStreamFromResource($symfonyRequest->getContent(true));
 
+        if (method_exists(Request::class, 'getContentTypeFormat')) {
+            $format = $symfonyRequest->getContentTypeFormat();
+        } else {
+            $format = $symfonyRequest->getContentType();
+        }
+
+        if (method_exists(Request::class, 'getPayload') && 'json' === $format) {
+            try {
+                $parsedBody = $symfonyRequest->getPayload()->all();
+            } catch (JsonException $e) {
+                $parsedBody = [];
+            }
+        } else {
+            $parsedBody = $symfonyRequest->request->all();
+        }
```

Comment:
```suggestion
        $format = $symfonyRequest->getContentType();
        if (method_exists(Request::class, 'getContentTypeFormat')) {
            $format = $symfonyRequest->getContentTypeFormat();
        }

        $parsedBody = $symfonyRequest->request->all();
        if (method_exists(Request::class, 'getPayload') && 'json' === $format) {
            try {
                $parsedBody = $symfonyRequest->getPayload()->all();
            } catch (JsonException $e) {
                $parsedBody = [];
            }
        }
```

What about to get rid of the `else`?

Reply:
The idea was to avoid the execution of ```$parsedBody = $symfonyRequest->request->all();``` when ```getPayload()``` is available

Reply:
The idea was to avoid the execution of $parsedBody = $symfonyRequest->request->all(); when getPayload() is available